R/connect.R

```diff
@@ -50,17 +50,17 @@ NULL
 #' }
 #' @export
 setMethod("dbConnect", "MySQLDriver",
-  function(drv, dbname = "", username = "", password = "", host = "",
-    unix.socket = "", port = 0, client.flag = 0,
-    groups = "rs-dbi", default.file = "", ...) {
+  function(drv, dbname = NULL, username = NULL, password = NULL, host = NULL,
+    unix.socket = NULL, port = 0, client.flag = 0,
+    groups = "rs-dbi", default.file = NULL, ...) {
 
     ptr <- connection_create(host, username, password, dbname, port, unix.socket,
       client.flag, groups, default.file)
 
     con <- new("MySQLConnection",
       ptr = ptr,
-      host = host,
-      db = dbname
+      host = if(is.null(host)) "" else host,
```

Comment:
Be aware that the `dbConnect()` arguments `host` and `dbname` can now be `NULL` again, while the [`MySQLConnection` object slots `host` and `db` are of class`character`][1]. Trying to put `NULL` in them leads to errors:

    Error in validObject(.Object) : 
    invalid class “MySQLConnection” object: 1: invalid object for slot "host" in class "MySQLConnection": got class "NULL", should be or extend class "character"
    invalid class “MySQLConnection” object: 2: invalid object for slot "db" in class "MySQLConnection": got class "NULL", should be or extend class "character"

That's why I put in this check.

[1]: https://github.com/rstats-db/RMySQL/blob/12ef4e38f8e6bac23cda97478e414552a76272a8/R/MySQLConnection.R#L12-L13